src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Windows.cs

```diff
@@ -57,7 +57,7 @@ internal ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipCo
                     {
                         bool closed = IsClosed;
                         bool alreadyBound = !IsInvalid && !IsClosed && (exception is ArgumentException);
-                        CloseAsIs(abortive: false);
+                        CloseAsIs(abortive: false, disposing: true);
```

Comment:
NIT: `disposing: true` looks weird here. I would flip the boolean in `CloseAsIs` signature, so this can be `finalizing: false` instead.